internal/status/gatewayclass.go

```diff
@@ -23,3 +25,25 @@ func SetGatewayClassAccepted(gc *gwapiv1.GatewayClass, accepted bool, reason, ms
 	gc.Status.Conditions = MergeConditions(gc.Status.Conditions, computeGatewayClassAcceptedCondition(gc, accepted, reason, msg))
 	return gc
 }
+
+// GetSupportedFeatures returns a list of supported Gateway-API features,
```

Comment:
recommend removing all the additions in the above files and inserting this inside `SetGatewayClassAccepted` in this file

Reply:
The function you recommended removing is mainly required for the tests, as we need to get the list of supported features. How would you recommend getting the supported features in tests if we remove this function?

Reply:
I actually meant https://github.com/envoyproxy/gateway/pull/2491/files#r1468272780, hope this helps